crates/sui-core/src/consensus_adapter.rs

```diff
@@ -1287,6 +1301,56 @@ impl SubmitToConsensus for Arc<ConsensusAdapter> {
         self.submit_batch(transactions, None, epoch_store)
             .map(|_| ())
     }
+
+    fn submit_best_effort(
+        &self,
+        transaction: &ConsensusTransaction,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+        // timeout is required, or the spawned task can run forever
```

Comment:
would the spawned task not be dropped at end of epoch bc of `within_alive_epoch`?

Reply:
if we consume all permits with infinitely retrying submit tasks, then the epoch may never end, because validators won't be able to sequence pending transactions